module/x/peggy/keeper/attestation_handler.go

```diff
@@ -0,0 +1,25 @@
+package keeper
+
+import (
+	"github.com/althea-net/peggy/module/x/peggy/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type AttestationHandler struct {
+	keeper Keeper
+}
+
+func (a AttestationHandler) Handle(ctx sdk.Context, att types.Attestation) error {
+	switch att.ClaimType {
+	case types.ClaimTypeEthereumBridgeDeposit:
+		// todo: mint new vouchers
+	case types.ClaimTypeEthereumBridgeWithdrawalBatch:
+		// todo: mark batch as successful
+	case types.ClaimTypeEthereumBridgeMultiSigUpdate:
+		// todo: update nonce for "MultiSig Set"
```

Comment:
also the other batch bookkeeping.

Reply:
I my head an orchestrator would submit two claims with a single message so that we have clear separation here. No implicit logic.
The ETH side should emit 2 events as well so that we keep this logically separated. Some (different) clients may only follow a subset of events.